src/com/owncloud/android/files/services/FileDownloader.java

```diff
@@ -97,6 +107,7 @@
     private NotificationCompat.Builder mNotificationBuilder;
     private int mLastPercent;
 
+    private static int jobId = 0;
```

Comment:
Here is the official info about it:

![captura de pantalla 2017-03-30 a las 9 53 39](https://cloud.githubusercontent.com/assets/15261363/24493340/cf1a083c-152e-11e7-90ff-cbc8ca783c67.png)

With this in mind, the approach I've used is not the proper one, because using the same id for different jobs implies overwriting other jobs, and therefore, not taking into account some of the failed uploads or downloads.

Other alternative may be using the download or upload file id, but this file id (if I'm not wrong) can be the same for different OC accounts so we have the same problem, we would overwrite jobs of other accounts.